diff -rupN SocketMan-v1.orig/Makefile SocketMan-v1/Makefile
--- SocketMan-v1.orig/Makefile	2018-12-12 10:44:29.721908693 +1000
+++ SocketMan-v1/Makefile	2018-12-12 10:51:09.393076351 +1000
@@ -3,7 +3,7 @@ IDIR = "./includes"
 PKG_SOURCE_VERSION:=$(COMMIT)
 INCLUDES = "-I/usr/include/libnl3"
 
-CFLAGS= -g -O2 -Wall -Wextra -Isrc -I$(IDIR) $(INCLUDES) $(CT_ARCH)
+CFLAGS+= -g -O2 -Wall -Wextra -Isrc -I$(IDIR) $(INCLUDES) $(CT_ARCH)
 
 ODIR=obj
 LDIR =../lib

